### ISSUE TITLE
AutoLink extension links invalid protocol-only (no host) URIs

### ISSUE THREAD

**Issue by peckjon** [atlassian/commonmark-java#99]
It is common to type only "http://" (not followed by any hostname etc) in the process of creating technical documentation.

The AutoLink extension incorrectly attempts to link these, resulting in ugly documentation.

URIs are invalid without a host portion:

```
URI = scheme ":" hier-part [ "?" query ] [ "#" fragment ]

hier-part = "//" authority path-abempty
                 / path-absolute
                 / path-rootless
                 / path-empty

authority = [ userinfo "@" ] host [ ":" port ]
```
- https://tools.ietf.org/html/rfc3986#appendix-A (brackets indicate optionality)

**Comment by robinst** [comment]
`http://` by itself should not get autolinked. Do you type `"http://"` (including the quotes)? If so, that's a bug in [autolink-java](https://github.com/robinst/autolink-java) which I should fix.

**Comment by peckjon** [comment]
Thanks for the quick response The line I typed was:

`* **(Required):** Images ("images": ["data://...", "http://...", ...])`

**Comment by robinst** [comment]
Thanks for including the full source :).

Yeah, I think it's valid because host can just be a single `unreserved` character. Having said that, even just `http://` is valid according to RFC 3986, but we don't autolink it. And for valid URLs that end with a `.`, we don't include `.` either.

I've raised https://github.com/robinst/autolink-java/issues/15 to fix it there, but we can also leave this one open until we bump the dependency here.

**Comment by peckjon** [comment]
Wow that was fast @robinst -- I see https://github.com/robinst/autolink-java/issues/15 is already fixed!

Thanks for being awesome.  Closing out.

**Comment by robinst** [comment]
Released in 0.10.0 now: https://github.com/atlassian/commonmark-java/blob/master/CHANGELOG.md#0100---2017-09-14

**Comment by peckjon** [comment]
Awesome, Robin! Thanks!!
